### PR TITLE
Craterlake staging

### PR DESCRIPTION
**Pull request by mposik1983**
### Briefly, what does this PR introduce?
Configures new tracking setup defined in craterlake.xml and craterlake_tracking_only.xml configurations
Screen Shot 2023-07-05 at 4 46 43 PM

### What kind of change does this PR introduce?
- [ ] Bug fix (issue #__)
- [ ] New feature (issue #__)
- [ ] Documentation update
- [x] Other:  New tracking configuration

### Please check if this PR fulfills the following:
- [ ] Tests for the changes have been added
- [ ] Documentation has been added / updated
- [x] Changes have been communicated to collaborators

### Does this PR introduce breaking changes? What changes might users need to make to their code?
This PR should not have changes that break other code. The new tracking configuration is included in a dedicated compact/tracking/definitions_craterlake.xml file

### Does this PR change default behavior?
Behavior should only change when craterlake config is selected. In which case tracking layers change and a few new MPGD layers are added.

### CODE REVIEW

**Comment by wdconinc** [comment]
Can you rebase this and drop commits already in main?

File: compact/tracking/tof_barrel.xml [hunk 1]
```diff
@@ -27,7 +27,7 @@
     <constant name="BarrelTOF_Module_thickness"       value="BarrelTOF_Sensor_thickness+2*BarrelTOF_Hybrid_thickness+2*BarrelTOF_CFSkin_thickness+BarrelTOF_CFoam_thickness+BarrelTOF_CoolingTube_thickness+BarrelTOF_Coolant_thickness"/>
 
     <constant name="BarrelTOF_zOffset"                 value="0*cm"/>
-    <constant name="BarrelTOF_radius_design"           value="64.6*cm"/>
+	    <!--<constant name="BarrelTOF_radius_design"           value="64.6*cm"/>-->
```
Comment (wdconinc) [inline compact/tracking/tof_barrel.xml:30]: Remove commented lines entirely.
```suggestion
```

File: configurations/brycecanyon.yml [hunk 2]
```diff
@@ -15,17 +15,7 @@ features:
     dirc:
     pfrich:
     drich:
-  ecal:
```
Comment (wdconinc) [inline configurations/brycecanyon.yml:18]: Don't modify brycecanyon.

File: compact/tracking/tof_endcap.xml [hunk 3]
```diff
@@ -9,8 +9,8 @@
       --------------------------
     </comment>
     <comment> Forward TOF position </comment>
-    <constant name="ForwardTOF_length"              value="min(ForwardTOFRegion_length, 2.*cm)"/>
-    <constant name="ForwardTOF_zmin"                value="ForwardTOFRegion_zmin + ForwardTOFRegion_length - ForwardTOF_length - 1*cm"/>
+	    <!--<constant name="ForwardTOF_length"              value="min(ForwardTOFRegion_length, 2.*cm)"/>
```
Comment (wdconinc) [inline compact/tracking/tof_endcap.xml:12]: Remove commented lines.

**Comment by mposik1983** [comment]
> 
Ok. I did rebase of the branch. I'm not sure how to drop commits though

File: src/BarrelPlanarMPGDTracker_geo.cpp [hunk 4]
```diff
@@ -320,4 +320,5 @@ static Ref_t create_MPGDDIRC_geo(Detector& description, xml_h e, SensitiveDetect
 
 //@}
 // clang-format off
-DECLARE_DETELEMENT(epic_MPGDDIRC, create_MPGDDIRC_geo)
+DECLARE_DETELEMENT(epic_OuterMPGDBarrel, create_BarrelPlanarMPGDTracker_geo)
+DECLARE_DETELEMENT(epic_MPGDDIRC, create_BarrelPlanarMPGDTracker_geo)
```
Comment (wdconinc) [inline src/BarrelPlanarMPGDTracker_geo.cpp:324]: Unused now(?).

File: src/MRich_geo.cpp [hunk 5]
```diff
@@ -1,465 +0,0 @@
-// SPDX-License-Identifier: LGPL-3.0-or-later
```
Comment (wdconinc) [inline src/MRich_geo.cpp:1]: Why remove Mrich?

**Review by wdconinc — CHANGES_REQUESTED** [review]

**Comment by wdconinc** [comment]
> Ok. I did rebase of the branch. I'm not sure how to drop commits though

Thanks. The commits in main disappeared.

File: src/MRich_geo.cpp (hunk 5 above)
Comment (wdconinc) [reply src/MRich_geo.cpp:1]: I mean why in this PR.

**Comment by wdconinc** [comment]
Won't be able to merge until checks return successfully. Can you figure out what is going wrong? Rebase might have solved some issues.

**Review by wdconinc — CHANGES_REQUESTED** [review]
We should avoid forward and backward as terminology. Recommendations (being discussed internally in the coordination team) are to use positive and negative endcap as is done for calorimeters. @mdiefent can share status. If we introduce new mpgd hit collections, they may want to use this terminology or would have to change anyway.

File: compact/tracking/tof_barrel.xml (hunk 1 above)
Comment (mposik1983) [reply compact/tracking/tof_barrel.xml]: Done.

File: configurations/brycecanyon.yml (hunk 2 above)
Comment (mposik1983) [reply configurations/brycecanyon.yml]: Done.

File: compact/tracking/tof_endcap.xml (hunk 3 above)
Comment (mposik1983) [reply compact/tracking/tof_endcap.xml]: Done

File: src/BarrelPlanarMPGDTracker_geo.cpp (hunk 4 above)
Comment (mposik1983) [reply src/BarrelPlanarMPGDTracker_geo.cpp:324]: This plug in is used by the new outer MPGD barrel and is now also used by the MPGD DIRC layer which was used in arches. Line 324 was left for comparability with arches config.

File: src/MRich_geo.cpp (hunk 5 above)
Comment (mposik1983) [reply src/MRich_geo.cpp]: I don't know. I am not sure how that happened. It's now back

File: configurations/craterlake.yml [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    vertex_barrel:
+    silicon_barrel:
+    mpgd_barrel:
+    mpgd_outerbarrel:
+    mpgd_forward_endcap:
+    mpgd_backward_endcap:
+    silicon_disks:
+    tof_barrel:
+    tof_endcap:
+  pid:
```
Comment (wdconinc) [inline configurations/craterlake.yml:16]: We want ecal, hcal in craterlake too.

**Comment by mposik1983** [comment]
> 

OK. When the decision is made on terminology, I can change the backward/forward terminology to whatever is decided.

File: compact/tracking/mpgd_outerbarrel.xml [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+    <constant name="MPGDOuterBarrelCathode_thickness"          value="MPGDOuterBarrelFoilKapton_thickness + MPGDOuterBarrelFoilCu_thickness"/>
+    <constant name="MPGDOuterBarrelFoil_thickness"             value="MPGDOuterBarrelFoilKapton_thickness + MPGDOuterBarrelFoilCu_thickness"/>
+    <constant name="MPGDOuterBarrelReadOut_thickness"          value="MPGDOuterBarrelReadOutNomex_thickness + MPGDOuterBarrelReadOutElectrode_thickness + MPGDOuterBarrelReadOutKapton_thickness"/>
+    <constant name="MPGDOuterBarrelModule_thickness"           value="MPGDOuterBarrelWindow_thickness + MPGDOuterBarrelCathode_thickness +
+                                                            MPGDOuterBarrelFoil_thickness + MPGDOuterBarrelReadOut_thickness +
+                                                            MPGDOuterBarrelPCB_thickness + MPGDOuterBarrelFrame_thickness" />
+  </define>
+
+
+  <materials>
```
Comment (wdconinc) [inline compact/tracking/mpgd_outerbarrel.xml:51]: Empty environments not needed.

File: compact/tracking/mpgd_outerbarrel.xml [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+    <constant name="MPGDOuterBarrelModule_allowed_space"           value="2.5*cm"/>
+    <constant name="MPGDOuterBarrelModule_rmax"                    value="MPGDOuterBarrelModule_rmin + MPGDOuterBarrelModule_allowed_space/2.0" />
+    <constant name="MPGDOuterBarrelModule_width"                   value="34.0*cm"/>
+    <constant name="MPGDOuterBarrelModule_length"                  value="0.5*(MPGDOuterBarrelModule_zmin1 + MPGDOuterBarrelModule_zmin2 + 2*MPGDOuterBarrelModule_allowed_space)"/>
+    <constant name="MPGDOuterBarrelModule_offset"                  value="(MPGDOuterBarrelModule_zmin2 - MPGDOuterBarrelModule_zmin1)/2.0"/>
+    <constant name="MPGDOuterBarrelModule_roverlap"                value="1.2*cm"/>
+    <constant name="MPGDOuterBarrelModule_zoverlap"                value="2*MPGDOuterBarrelFrame_width"/>
+
+
+    <comment> layer parameters </comment>
```
Comment (wdconinc) [inline compact/tracking/mpgd_outerbarrel.xml:30]:
```suggestion
    <comment> Layer parameters </comment>
```

File: compact/tracking/mpgd_outerbarrel.xml [hunk 9]
```diff
@@ -0,0 +1,195 @@
...
+      version="1.0"
+><comment/>
+</info>
+
+<define>
+    <comment> frames </comment>
+    <constant name="MPGDOuterBarrelFrame_width"            value="20*mm"/>
+    <constant name="MPGDOuterBarrelFrame_thickness"        value="7*mm"/>
+
+    <comment> module constants </comment>
```
Comment (wdconinc) [inline compact/tracking/mpgd_outerbarrel.xml:19]:
```suggestion
    <comment> Module constants </comment>
```

File: compact/tracking/mpgd_outerbarrel.xml [hunk 10]
```diff
@@ -0,0 +1,195 @@
...
+      title="Micro-Pattern Gas Detector Planar Barrel Layer"
+      author="mposik1983"
+      url="https://github.com/mposik1983"
+      status="development"
+      version="1.0"
+><comment/>
+</info>
+
+<define>
+    <comment> frames </comment>
```
Comment (wdconinc) [inline compact/tracking/mpgd_outerbarrel.xml:15]:
```suggestion
    <comment> Frames </comment>
```

File: compact/tracking/mpgd_barrel.xml [hunk 11]
```diff
@@ -23,7 +23,9 @@
     <constant name="InnerMPGDBarrelMod_rmin"                value="InnerMPGDBarrel_rmin"/>
     <constant name="InnerMPGDBarrelMod_dz"                  value="InnerMPGDBarrel_dz"/>
 
-    <constant name="InnerMPGDBarrelMod_length"              value="2 * InnerMPGDBarrelMod_rmin / tan(TrackerBackwardAngle) - InnerMPGDBarrel_dz" />
+	    <!--<constant name="InnerMPGDBarrelMod_length"              value="2 * InnerMPGDBarrelMod_rmin / tan(TrackerBackwardAngle) - InnerMPGDBarrel_dz" />-->
```
Comment (wdconinc) [inline compact/tracking/mpgd_barrel.xml:26]: Remove comment.

File: compact/tracking/definitions_craterlake.xml [hunk 12]
```diff
@@ -0,0 +1,209 @@
+<!-- SPDX-License-Identifier: LGPL-3.0-or-later -->
+<!-- Copyright (C) 2022 Sylvester Joosten, Shujie Li -->
+
+<lccdd>
+
+  <display>
```
Comment (wdconinc) [inline compact/tracking/definitions_craterlake.xml:6]: Remove unused environment.

File: compact/tracking/definitions_craterlake.xml [hunk 13]
```diff
@@ -0,0 +1,209 @@
...
+    <comment> The forward outer cone is projective from the starting rmin1 </comment>
+    <constant name="ForwardEndcapSupportCone_rmin1"  value="85*cm" />
+    <constant name="ForwardEndcapSupportCone_zmax"   value="190*cm" />
+    <comment> Discontinuity for the cone to the cylinder to avoid overlaps FIXME </comment>
+    <constant name="BackwardEndcapSupportCyl_rmin"   value="65*cm" />
+    <constant name="BackwardEndcapSupportCyl_zmin"   value="BackwardPIDRegion_zmin + 5*cm" />
+    <constant name="BackwardEndcapSupportCyl_zmax"   value="BackwardPIDRegion_zmax" />
+
+    <comment> Main parameters for the positive silicon disks </comment>
+    <comment> assume the disk are centered along z. The inner radii are larger than designed to account for the beampipe offset. Need to be fixed later </comment>
```
Comment (wdconinc) [inline compact/tracking/definitions_craterlake.xml:73]:
```suggestion
    <comment> Assume the disk are centered along z. The inner radii are larger than designed to account for the beampipe offset. Need to be fixed later </comment>
```

File: compact/tracking/definitions_craterlake.xml [hunk 14]
```diff
@@ -0,0 +1,209 @@
...
+    <constant name="TrackerEndcapNDisk2_rmin"        value="Beampipe_rmax + Beampipe_bakeout_buffer"   />
+    <constant name="TrackerEndcapNDisk2_rmax"        value="TrackerEndcapPDisk2_rmax" />
+    <constant name="TrackerEndcapNDisk3_zmin"        value="85*cm" />
+    <constant name="TrackerEndcapNDisk3_rmin"        value="34.244*mm + Beampipe_bakeout_buffer + 0.817*mm" />
+    <constant name="TrackerEndcapNDisk3_rmax"        value="TrackerEndcapPDisk3_rmax" />
+    <constant name="TrackerEndcapNDisk4_zmin"        value="105*cm" />
+    <constant name="TrackerEndcapNDisk4_rmin"        value="38.043*mm + Beampipe_bakeout_buffer + 3.310*mm" />
+    <constant name="TrackerEndcapNDisk4_rmax"        value="TrackerEndcapPDisk4_rmax" />
+
+    <comment> Main paratmers for MPGD endcap disks, offset here is the distance between disks </comment>
```
Comment (wdconinc) [inline compact/tracking/definitions_craterlake.xml:110]:
```suggestion
    <comment> Main parameters for MPGD endcap disks, offset here is the distance between disks </comment>
```

File: compact/tracking/definitions_arches.xml [hunk 15]
```diff
@@ -136,7 +156,6 @@
         type="DD4hep_SubdetectorAssembly"
         vis="TrackerSubAssemblyVis">
         <composite name="MPGDDIRC_0"/>
-    </detector>
```
Comment (wdconinc) [inline compact/tracking/definitions_arches.xml:139]: ?

File: compact/definitions.xml [hunk 16]
```diff
@@ -442,18 +442,10 @@ Examples:
...
-    <constant name="ForwardTOFRegion_minR"            value="8*cm" />
-    <constant name="ForwardTOFRegion_maxR"            value="67*cm" />
-    <constant name="ForwardTOF_xOffset"         value="-2.75*cm" />
-    <constant name="ForwardTOF_det_height"      value="2.0*cm" />
 
     <comment> Forward RICH region </comment>
-    <constant name="ForwardRICHRegion_zmin"       value="ForwardTOFRegion_zmin + ForwardTOFRegion_length" />
+    <constant name="ForwardRICHRegion_zmin"       value="ForwardPIDRegion_zmin + ForwardTOFRegion_length" />
     <constant name="ForwardRICHRegion_length"     value="ForwardPIDRegion_length - ForwardTOFRegion_length" />
     <comment> tangent of the opening angle at the front and back of the RICH </comment>
```
Comment (wdconinc) [inline compact/definitions.xml:450]:
```suggestion
     <comment> Tangent of the opening angle at the front and back of the RICH </comment>
```

**Review by wdconinc — CHANGES_REQUESTED** [review]

File: configurations/craterlake.yml (hunk 6 above)
Comment (mposik1983) [reply configurations/craterlake.yml:16]: Added.

File: compact/tracking/mpgd_outerbarrel.xml (hunk 7 above)
Comment (mposik1983) [reply compact/tracking/mpgd_outerbarrel.xml]: Removed

File: configurations/craterlake.yml [hunk 17]
```diff
@@ -0,0 +1,34 @@
...
+    dirc:
+    pfrich:
+    drich:
+  ecal:
+   forward_homogeneous:
```
Comment (wdconinc) [inline configurations/craterlake.yml:21]:
```suggestion
    forward_homogeneous:
```

**Review by wdconinc — CHANGES_REQUESTED** [review]

File: configurations/craterlake.yml (hunk 17 above)
Comment (wdconinc) [reply configurations/craterlake.yml:21]: Indentation matters in yaml.

File: compact/tracking/mpgd_barrel.xml (hunk 11 above)
Comment (mposik1983) [reply compact/tracking/mpgd_barrel.xml]: done

**Review by wdconinc — APPROVED** [review]

**Pull request merged**